packages/evergreen-component-mapper/src/text-field/text-field.tsx

```diff
@@ -1,25 +1,34 @@
 import React from 'react';
 
-import { TextInputField } from 'evergreen-ui';
+import { Autocomplete, TextInputField } from 'evergreen-ui';
 
 import useFieldApi from '@data-driven-forms/react-form-renderer/use-field-api';
 import { UseFieldApiProps } from '@data-driven-forms/react-form-renderer';
 
 export interface TextFieldProps extends UseFieldApiProps<string> {
-    name: string;
-    isRequired?: boolean;
-};
+	name: string;
+	items?: string[];
+	isRequired?: boolean;
+}
 
 const TextField: React.FC<TextFieldProps> = (props) => {
-	const { input, meta, isRequired, ...rest } = useFieldApi(props);
+	const { input, meta, isRequired, items, ...rest } = useFieldApi(props) as TextFieldProps;
```

Comment:
I think a generic can be used here:

```ts
const { input, meta, isRequired, items, ...rest } = useFieldApi<TextFieldProps>(props);
```

Reply:
So the reason I've used the type assertion here instead of the generic type was because that TypeScript would fail to infer the type of the `props` object that was returned from this hook. If you look closely into the return type of this hook and dig deeper, you can find that the first generic type parameter used for the hook ultimately just sets the type of the `value` in the `FieldInputProps` interface and does not set or shape the return type of this hook.
For your reference:
```typescript
export interface UseFieldApiProps<
 FieldValue,
T extends HTMLElement = HTMLElement> extends AnyObject {
  input: FieldInputProps<FieldValue, T>;
  meta: Meta<FieldValue>;
}

export default function<T = any>(options: UseFieldApiConfig): UseFieldApiProps<T>;

interface FieldInputProps<FieldValue, T extends HTMLElement = HTMLElement>
  extends AnyObject {
  name: string;
  onBlur: (event?: React.FocusEvent<T>) => void;
  onChange: (event: React.ChangeEvent<T> | any) => void;
  onFocus: (event?: React.FocusEvent<T>) => void;
  type?: string;
  value: FieldValue;
  checked?: boolean;
  multiple?: boolean;
}
```
Above was copied from the exported type files from DDF repo.

However, from my usages of this hook in my other projects, I know for sure that what this hook returns usually is the same in structure (with some extra things added by this hook) as the provided `props` object here.

In my opinion, for the scope of this PR, I believe we should leave the type assertion here for now for better type inference. Later, something can be done about making the first generic type parameter to actually shape the return type of this hook.

Reply:
Oh, Thank you for so detailed response!